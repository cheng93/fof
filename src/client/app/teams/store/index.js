import { actions } from './teams.actions';
import { getters } from './teams.getters';
import { mutations } from './teams.mutations';
import { state } from './teams.state';

export default {
    namespaced: true,
    actions,
    getters,
    mutations,
    state
};
