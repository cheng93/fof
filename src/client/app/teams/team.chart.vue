<script>
import numeral from 'numeral';
import palette from 'google-palette';
import { mapState } from 'vuex';
import { Bar } from 'vue-chartjs';

export default {
    computed: mapState({
        labels: state => state.teams.teamData.seasons.map(s => s.year),
        datasets(state) {
            var colours = palette('tol-rainbow', 3).map(function(hex) {
                return '#' + hex;
            });
            var keys = {
                win_lose_percent: {
                    label: 'WL%',
                    type: 'line',
                    map(s) {
                        return numeral(s.win_lose_percent * 100).format('0.00');
                    },
                    borderColor: colours[0],
                    fill: false,
                    yAxisID: 'B'
                },
                wins: {
                    label: 'Wins',
                    backgroundColor: colours[1],
                    yAxisID: 'A'
                },
                loses: {
                    label: 'loses',
                    backgroundColor: colours[2],
                    yAxisID: 'A'
                }
            };
            return Object.entries(keys).map(([key, option]) => ({
                ...option,
                data: state.teams.teamData.seasons.map(
                    s => (option.map && option.map(s)) || s[key]
                )
            }));
        }
    }),
    extends: Bar,
    mounted() {
        // Overwriting base render method with actual data.
        let data = {
            labels: this.labels,
            datasets: this.datasets
        };
        let options = {
            maintainAspectRatio: false,
            scales: {
                yAxes: [
                    {
                        id: 'A',
                        type: 'linear',
                        position: 'left',
                        gridLines: {
                            display: false
                        }
                    },
                    {
                        id: 'B',
                        type: 'linear',
                        gridLines: {
                            display: false
                        },
                        position: 'right',
                        ticks: {
                            min: 0,
                            max: 100
                        }
                    }
                ]
            }
        };
        this.renderChart(data, options);
    }
};
</script>
