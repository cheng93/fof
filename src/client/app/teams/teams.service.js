import axios from 'axios';

export default {
    getTeam(teamId) {
        return axios.get(`api/teams/${teamId}`);
    },
    getTeams() {
        return axios.get('api/teams');
    }
};
