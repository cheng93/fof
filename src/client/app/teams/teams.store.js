import teamsService from './teams.service';

export const actionTypes = {
    GET_TEAMS: 'GET_TEAMS'
};

export const mutationTypes = {
    SET_TEAMS: 'SET_TEAMS'
};

const state = {
    teams: []
};

const getters = {};

const actions = {
    [actionTypes.GET_TEAMS]({ commit }) {
        teamsService
            .getTeams()
            .then(response =>
                commit(mutationTypes.SET_TEAMS, response.data.teams)
            );
    }
};

const mutations = {
    [mutationTypes.SET_TEAMS](state, teams) {
        state.teams = teams;
    }
};

export default {
    namespaced: true,
    state,
    getters,
    actions,
    mutations
};
