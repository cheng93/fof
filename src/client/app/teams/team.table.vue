<template>
    <v-data-table
        :headers="headers"
        :items="seasons"
        hide-actions
        class="elevation-1">
        <template slot="items" slot-scope="props">
            <td>
                {{props.item.year}}
            </td>
            <td>{{ props.item.wins }}</td>
            <td>{{ props.item.loses }}</td>
            <td>{{ props.item.ties }}</td>
            <td>{{ props.item.win_lose_percent | to_percentage }}</td>
            <td>{{ props.item.standing_name }}</td>
            </template>
    </v-data-table>
</template>
<script>
import { mapState } from 'vuex';
import { actionTypes } from './store/teams.actions';

export default {
    computed: mapState({
        seasons: state => state.teams.teamData.seasons
    }),
    data() {
        return {
            headers: [
                {
                    text: 'Season',
                    value: 'year'
                },
                {
                    text: 'Wins',
                    value: 'wins'
                },
                {
                    text: 'Loses',
                    value: 'loses'
                },
                {
                    text: 'Ties',
                    value: 'ties'
                },
                {
                    text: 'WL%',
                    value: 'win_lose_percent'
                },
                {
                    text: 'Standing',
                    value: 'standing_name'
                }
            ]
        };
    }
};
</script>

