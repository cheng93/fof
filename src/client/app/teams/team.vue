<template>
    <section>
        <h1 class="mdc-typography--headline5">{{team.name}}</h1>
        <dl>
            <dt class="mdc-typography--subtitle2">
                Division:
            </dt>
            <dd class="mdc-typography--body2">
                {{team.division}}
            </dd>
            <dt class="mdc-typography--subtitle2">
                Record (W-L-T):
            </dt>
            <dd class="mdc-typography--body2">
                {{team.wins}}-{{team.loses}}-{{team.ties}}
            </dd>
            <dt class="mdc-typography--subtitle2">
                Win-Lose %:
            </dt>
            <dd class="mdc-typography--body2">
                {{team.win_lose_percent | to_percentage}}
            </dd>
            <dt class="mdc-typography--subtitle2">
                Playoff Appearances:
            </dt>
            <dd class="mdc-typography--body2">
                {{team.playoff_appearances}}
            </dd>
            <dt class="mdc-typography--subtitle2">
                Conference Wins:
            </dt>
            <dd class="mdc-typography--body2">
                {{team.conference_wins}}
            </dd>
            <dt class="mdc-typography--subtitle2">
                Superbowl Wins:
            </dt>
            <dd class="mdc-typography--body2">
                {{team.superbowl_wins}}
            </dd>
        </dl>
        <tabs :headers="headers">
            <template slot="tab-0">
                <team-table></team-table>
            </template>
            <template slot="tab-1">
                Hello World
            </template>
        </tabs>
    </section>
</template>
<script>
import { mapState } from 'vuex';
import { actionTypes } from './store/teams.actions';
import store from '../store';

import tabs from '../components/tabs/tabs';
import teamTable from './team.table';

export default {
    computed: mapState({
        team: state => state.teams.teamData
    }),
    beforeRouteEnter(to, from, next) {
        store
            .dispatch(`teams/${actionTypes.GET_TEAM}`, to.params.id)
            .then(() => next());
    },
    data() {
        return {
            headers: ['Table', 'Chart']
        };
    },
    components: {
        tabs,
        teamTable
    }
};
</script>
<style lang="scss" scoped>
h1,
dl {
    margin: 12px 0;
}
dl {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    max-width: 248px;

    dt {
        flex: 1 1 124px;
    }
}
</style>

