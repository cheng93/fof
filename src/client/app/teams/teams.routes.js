import teams from './teams';

const routes = [{ path: '/teams', component: teams }];

export default routes;
