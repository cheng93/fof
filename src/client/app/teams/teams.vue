<template>
    <section>
        <h1 class="mdc-typography--headline5">Teams</h1>
        <v-data-table
            :headers="headers"
            :items="teams"
            hide-actions
            class="elevation-1">
             <template slot="items" slot-scope="props">
                <td>{{ props.item.name }}</td>
                <td>{{ props.item.wins }}</td>
                <td>{{ props.item.loses }}</td>
                <td>{{ props.item.ties }}</td>
                <td>{{ props.item.win_lose_percent | win_lose_percent }}</td>
                <td>{{ props.item.playoff_appearances }}</td>
                <td>{{ props.item.conference_wins }}</td>
                <td>{{ props.item.superbowl_wins }}</td>
                </template>
        </v-data-table>
    </section>
</template>
<script>
import numeral from 'numeral';
import { mapState } from 'vuex';
import { actionTypes } from './teams.store';

export default {
    computed: mapState({
        teams: state => state.teams.teams
    }),
    created() {
        this.$store.dispatch(`teams/${actionTypes.GET_TEAMS}`);
    },
    data() {
        return {
            headers: [
                {
                    text: 'Team',
                    value: 'name'
                },
                {
                    text: 'Wins',
                    value: 'wins'
                },
                {
                    text: 'Loses',
                    value: 'loses'
                },
                {
                    text: 'Ties',
                    value: 'ties'
                },
                {
                    text: 'WL%',
                    value: 'win_lose_percent'
                },
                {
                    text: 'Playoffs',
                    value: 'playoff_appearances'
                },
                {
                    text: 'Conference',
                    value: 'conference_wins'
                },
                {
                    text: 'Superbowl',
                    value: 'superbowl_wins'
                }
            ]
        };
    },
    filters: {
        win_lose_percent: function(value) {
            return `${numeral(value * 100).format('0.00')}%`;
        }
    }
};
</script>

<style scoped>
h1 {
    margin: 12px 0;
}

ul {
    display: flex;
    flex-wrap: wrap;
}

li {
    flex: 0 1 200px;
    margin: 12px 0;
}
</style>

