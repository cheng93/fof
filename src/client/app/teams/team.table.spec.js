import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import { actionTypes } from './store/teams.actions';
import teamTable from './team.table.vue';

describe('Team Table', () => {
    let wrapper;

    beforeEach(() => {
        const localVue = createLocalVue();
        localVue.use(Vuex);
        localVue.use(Vuetify);
        const actions = {
            [actionTypes.GET_TEAM]: jest.fn()
        };

        const store = new Vuex.Store({
            modules: {
                teams: {
                    namespaced: true,
                    actions
                }
            }
        });

        const seasons = [
            {
                year: 2015,
                wins: 2,
                loses: 1,
                ties: 1,
                win_lose_percent: 0.3333,
                standing_name: 'Regular'
            },
            {
                year: 2016,
                wins: 3,
                loses: 0,
                ties: 0,
                win_lose_percent: 1,
                standing_name: 'Runner Up'
            }
        ];

        const $route = {
            params: {
                id: 1
            }
        };

        wrapper = mount(teamTable, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            computed: {
                seasons: () => seasons
            },
            filters: {
                to_percentage: val => `toPercentage(${val})`
            },
            mocks: {
                $route
            },
            store,
            localVue
        });
    });

    it('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
