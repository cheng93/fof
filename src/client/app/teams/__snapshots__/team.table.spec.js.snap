// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Table matches snapshot 1`] = `
<div class="elevation-1">
  <div class="v-table__overflow">
    <table class="v-datatable v-table">
      <thead>
        <tr>
          <th role="columnheader" scope="col" aria-label="Season: Sorted ascending. Activate to sort descending." aria-sort="ascending" tabindex="0" class="column sortable active asc text-xs-left">Season
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
          <th role="columnheader" scope="col" aria-label="Wins: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Wins
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
          <th role="columnheader" scope="col" aria-label="Loses: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Loses
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
          <th role="columnheader" scope="col" aria-label="Ties: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Ties
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
          <th role="columnheader" scope="col" aria-label="WL%: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">WL%
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
          <th role="columnheader" scope="col" aria-label="Standing: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Standing
            <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
          </th>
        </tr>
        <tr class="v-datatable__progress">
          <th colspan="6" class="column"></th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            2015
          </td>
          <td>2</td>
          <td>1</td>
          <td>1</td>
          <td>toPercentage(0.3333)</td>
          <td>Regular</td>
        </tr>
        <tr>
          <td>
            2016
          </td>
          <td>3</td>
          <td>0</td>
          <td>0</td>
          <td>toPercentage(1)</td>
          <td>Runner Up</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
